src/backend/ipcHandlers/mods.ts

```diff
@@ -0,0 +1,306 @@
+
+  sendFrontendMessage('gameStatusUpdate', {
+    appName,
+    status: 'extracting',
+    runner: 'hyperplay',
+    folder: dirPath
+  })
+
+  window.webContents.send(`progressUpdate-${appName}`, {
+    appName,
+    runner: 'hyperplay',
+    folder: dirPath,
+    status: 'extracting',
+    progress: {
+      folder: dirPath,
+      percent: 0,
+      diskSpeed: 0,
+      downSpeed: 0,
+      bytes: 0,
+      eta: null
+    }
+  })
+
+  logInfo(
+    `Extracting ${zipFile} to ${dirPath} -----------------`,
+    LogPrefix.HyperPlay
+  )
+  const extractService = new ExtractZipService(zipFile, dirPath, {
+    deleteOnEnd: false
+  })
+
+  inProgressExtractionsMap.set(appName, extractService)
+  const extractedFolder = path
+    .basename(extractService.source)
+    .replace('.zip', '')
+  const extractedFolderFullPath = path.join(dirPath, extractedFolder)
+
+  await new Promise((resolve) => {
+    extractService.on(
+      'progress',
+      ({
+        processedSizeInBytes,
+        totalSizeInBytes,
+        speedInBytesPerSec,
+        progressPercentage
+      }: ExtractZipProgressResponse) => {
+        logInfo(
+          `Extracting Progress: ${progressPercentage}% Speed: ${speedInBytesPerSec} B/s | Total size ${totalSizeInBytes} and ${processedSizeInBytes}`,
+          LogPrefix.HyperPlay
+        )
+        const currentProgress = calculateProgress(
+          processedSizeInBytes,
+          totalSizeInBytes,
+          speedInBytesPerSec,
+          speedInBytesPerSec,
+          progressPercentage
+        )
+
+        window.webContents.send(`progressUpdate-${appName}`, {
+          appName,
+          runner: 'hyperplay',
+          folder: dirPath,
+          status: 'extracting',
+          progress: {
+            folder: dirPath,
+            ...currentProgress
+          }
+        })
+      }
+    )
+
+    extractService.on('canceled', () => {
+      logInfo(`Canceled Extracting of base game file`, LogPrefix.HyperPlay)
+
+      process.noAsar = false
+
+      cancelQueueExtraction()
+      callAbortController(appName)
+
+      sendFrontendMessage('gameStatusUpdate', {
+        appName,
+        status: 'done',
+        runner: 'hyperplay',
+        folder: dirPath
+      })
+
+      window.webContents.send(`progressUpdate-${appName}`, {
+        appName,
+        runner: 'hyperplay',
+        folder: dirPath,
+        status: 'done',
+        progress: {
+          folder: dirPath,
+          percent: 0,
+          diskSpeed: 0,
+          downSpeed: 0,
+          bytes: 0,
+          eta: null
+        }
+      })
+
+      notify({
+        title: i18next.t('mod.baseGame.cancel.title', 'Canceled Extraction'),
+        body: i18next.t(
+          'mod.baseGame.cancel.body',
+          'Base game extraction canceled'
+        )
+      })
+
+      sendFrontendMessage('refreshLibrary', 'hyperplay')
+      try {
+        rm(
+          extractedFolderFullPath,
+          { recursive: true, force: true },
```

Comment:
Can we add a TODO comment here summarizing the prior work or linking to the windows issue on node.js repo?

Reply:
I'll be updating this issue on our end as well https://github.com/HyperPlay-Gaming/product-management/issues/608

Reply:
thanks for creating the issue so we can take a look at it later.
So far I can confirm that this is not a bug on NodeJS anymore, it was fixed on node v14, that is why it works when uninstalling the games. 
I believe that the problem might be on our extract service that sometimes doesn't release one of the files (common2.MPQ in my case) and Windows won't allow removing it, since it is in use. We should refactor the extract service to immediately release the files when cancelling. 
But it requires further investigation.